hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -447,7 +450,14 @@ public void validateTableProperties(Properties properties) {
 
     // Meta fields can be disabled only when either {@code SimpleKeyGenerator}, {@code ComplexKeyGenerator}, {@code NonpartitionedKeyGenerator} is used
     if (!getTableConfig().populateMetaFields()) {
-      String keyGenClass = properties.getProperty(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME.key(), "org.apache.hudi.keygen.SimpleKeyGenerator");
+      String keyGenClass;
+      if (properties.containsKey(HoodieTableConfig.KEY_GENERATOR_TYPE.key())) {
+        keyGenClass = KeyGeneratorType.valueOf(properties.getProperty(HoodieTableConfig.KEY_GENERATOR_TYPE.key())).getClassName();
+      } else if (properties.containsKey(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME.key())) {
+        keyGenClass = properties.getProperty(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME.key());
+      } else {
+        keyGenClass = "org.apache.hudi.keygen.SimpleKeyGenerator";
```

Comment:
Move this into `KeyGeneratorType`

Reply:
Can we also move this constant class name "org.apache.hudi.keygen.SimpleKeyGenerator" into `KeyGeneratorType`.